### PR TITLE
Added note about imports where necessary

### LINKED ISSUE #12936: Indicate in documentation when something has to imported via require
## Discussion
Currently some parts of the standard library must be imported via `require x` before usage. Whatever the reason for this, it leads to some usage problems:
1. It seems arbitrary which ones must be imported (there may be an internal reason, but it's not clear to a user)
2. It is not clear what is to imported (JSON is `json` but BigInt is just `big`?)

And it is never specified anywhere in the api documentation or crystal book, what is to be imported and why. In some cases it is shown in the example case near the top, in others such as [BigInt](https://crystal-lang.org/api/1.7.0/BigInt.html) not even that. In any case, simply having it in an example is very inexplicit. 

In light of this I suggest:
1. Every class/module/whatever that needs an import should state so near the top
2. A section should be added to the crystal book and/or api homepage explaining how to use the standard library and that `require` may be needed sometimes

I am not sure about the style. The ruby docs have a heading dedicated to it ([example](https://docs.ruby-lang.org/en/3.2/JSON.html)) but perhaps a NOTE would be better.

### PR DESCRIPTION
**Pull request by Tamnac**
Closes #12936 

I tried my best to figure out the appropriate thing to import, but it turned out more complicated than expected, so may not be perfect.

Uses a NOTE in the overview section of each page.

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

File: src/big/big_float.cr [hunk 1]
```diff
@@ -4,6 +4,9 @@ require "big"
 # A `BigFloat` can represent arbitrarily large floats.
 #
 # It is implemented under the hood with [GMP](https://gmplib.org/).
+#
+# NOTE: To use `BigFloat`, you must explicitly import it with `require "big"`
+#
```
Comment (HertzDevil) [inline src/big/big_float.cr:9]: unnecessary trailing empty comment line (there are quite a few of these)

File: src/colorize.cr [hunk 2]
```diff
@@ -3,6 +3,9 @@
 # as its main interface, which calls `to_s` and surrounds it with the necessary escape codes
 # when it comes to obtaining a string representation of the object.
 #
+# NOTE: To use `Colorize`, you must explicitly import it with `require "colorize"`
+#
```
Comment (HertzDevil) [inline src/colorize.cr:7]: unnecessary double empty comment line (there are also quite a few of these)

File: src/ecr.cr [hunk 3]
```diff
@@ -18,6 +18,8 @@
 #
 # Quick Example:
 #
+# NOTE: To use `ECR`, you must explicitly import it with `require "ecr"`
```
Comment (HertzDevil) [inline src/ecr.cr:21]: This placement right after the `Quick Example:` is a bit odd

File: src/log.cr [hunk 4]
```diff
@@ -18,7 +20,7 @@
 # ```
 # Log.info &.emit("User logged in", user_id: 42)
 # ```
-#
+#
```
Comment (HertzDevil) [inline src/log.cr:23]: This space and presumably some others won't pass the formatter

File: src/yaml.cr [hunk 5]
```diff
@@ -10,6 +10,8 @@ require "base64"
 # version 1.1 to/from native Crystal data structures, with the additional
 # independent types specified in http://yaml.org/type/
 #
+# NOTE: To use `YML`, you must explicitly import it with `require "yml"`
```
Comment (oprypin) [inline src/yaml.cr:13]: typo "yaml" **twice**

**Review by HertzDevil — APPROVED** [review]

**Comment by Sija** [comment]
I'd rather use term `require`, since `import` is sth non-existent in Crystal, so far and it could be easily mistaken with import features in other languages, which are quite different.

**Comment by HertzDevil** [comment]
The terminology is appropriate though, since you only require a file, not a type.

**Comment by straight-shoota** [comment]
Thanks for raising this issue. But I think it's okay.
The exact term is not critical.

**Comment by Sija** [comment]
> The terminology is appropriate though, since you only require a file, not a type.

Yeah, but the word in question is _import_.

**Pull request merged**